### ISSUE TITLE
Resource Tracking improvements

### ISSUE THREAD

**Issue by SCA-ZMT** [ITISFoundation/osparc-issues#1307]

**Comment by matusdrobuliak66** [comment]
# Performance Analysis

- db.t3.medium (2 vCPU, 4GB RAM)

| Rows       | Statement   | Execution time (ms) | CSV size |
| ---------- | ----------- | ------------------- | -------- |
| 1 000 000  | sum credits | 290.928             |          |
| 1 000 000  | list (join) | 2856.100            |          |
| 1 000 000  | export      | 9692.535            | 192 MB   |
| ---------- | ----------- | ------------------- | -------- |
| 5 000 000  | sum credits | 2025.984            |          |
| 5 000 000  | list (join) | 37840.759           |          |
| 5 000 000  | export      | 83072.873           | 1 GB     |
| ---------- | ----------- | ------------------- | -------- |
| 10 000 000 | sum credits | 11621.637           |          |
| 10 000 000 | list (join) | 90805.676           |          |
| 10 000 000 | export      | 224646.016          | 2.1 GB   |

- db.t3.xlarge (4 vCPU, 16GB RAM)

| Rows       | Statement   | Execution time (ms) | CSV size |
| ---------- | ----------- | ------------------- | -------- |
| 1 000 000  | sum credits | 312.325             |          |
| 1 000 000  | list (join) | 1424.304            |          |
| 1 000 000  | export      | 11081.482           | 192 MB   |
| ---------- | ----------- | ------------------- | -------- |
| 5 000 000  | sum credits | 1247.781            |          |
| 5 000 000  | list (join) | 28066.748           |          |
| 5 000 000  | export      | 41154.578           | 1 GB     |
| ---------- | ----------- | ------------------- | -------- |
| 10 000 000 | sum credits | 11871.515           |          |
| 10 000 000 | list (join) | 30492.221           |          |
| 10 000 000 | export      | 86182.641           | 2.1 GB   |

**Comment by sanderegg** [comment]
some questions from our test together:
- what are the implications on the database responsiveness when you run these queries?
- we saw that the frontend was blocked when running these queries (with some taking a very long time, and other failing), did you find out why?

**Comment by matusdrobuliak66** [comment]
> some questions from our test together:
> 
> * what are the implications on the database responsiveness when you run these queries?
> * we saw that the frontend was blocked when running these queries (with some taking a very long time, and other failing), did you find out why?

Yes, when the frontend lists usage endpoint, we join 2 tables in the database and filter/sort based on user requests. When this was called multiple times, it caused the database to be overloaded, resulting in timeouts of requests as the database was unable to process all the queries at once.

1. https://github.com/ITISFoundation/osparc-simcore/pull/5720 should optimize the query itself.
2. We discussed with Odei that in the long term, we might change the logic of the endpoint and allow users to see only previews. This means the frontend will not overload the backend with so many queries. If a user wants more, they will need to use the export functionality.
3. Also, as you can see from the observation, we are using the database with only 2 CPUs. When I increased the instance to 4 CPUs, the performance of this query improved significantly. With this, we could also process more queries in parallel.

**Comment by mrnicegyu11** [comment]
after talking to @matusdrobuliak66 we can close this parent case, keeping the linked issue around.
